#  1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


     Minitab File : Cutlets.mtw

This is a 2 sample t test

Here, 

H0 : unitA=unitB ( no diff with 2 units)

H1 : unit!=unitB ( diff with 2 units )

In [2]:
import pandas as pd
from scipy import stats

In [4]:
data = pd.read_csv("Cutlets.csv")

In [3]:
data

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [5]:
unit_A=data.iloc[:,0]
unit_B=data.iloc[:,1]

In [6]:
unit_A

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [7]:
unit_B

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [8]:
stats.ttest_ind(unit_A,unit_B)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

0.47>0.05


Accept null hypothesis.

That is,

There is no difference between unitA and unit B

# 2.  A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


Here 2 or more groups are present, so we use Anova Test.

H0 : Lab1=Lab2=Lab3=Lab4
    
H1 : Lab1!=Lab2!=Lab3!=Lab4

In [5]:
data = pd.read_csv("LabTAT.csv")

In [10]:
data

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [16]:
lab1=data.iloc[:,0]
lab2=data.iloc[:,1]
lab3=data.iloc[:,2]
lab4=data.iloc[:,3]

In [19]:

stats.f_oneway(lab1,lab2,lab3,lab4)

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Here,

2.e-57 < 0.05

So we reject H0.

Accept H1.

So 4 laboratories are not equal.

# 3. Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


Here categorical data are present. So we use Chi square test.

Ho : Variables are dependent.

H1 : Variables are independent.

In [6]:
data = pd.read_csv("BuyerRatio.csv")

In [37]:
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [34]:
east = data.iloc[:,1]
west = data.iloc[:,2]
north = data.iloc[:,3]
south = data.iloc[:,4]

In [39]:
stats.chi2_contingency([east,west,north,south])


(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

The p value is 0.66

0.66 >0.05

Accept H0.

ie, Varibles are dependent.

There is similar male-female buyer ratios.

# 4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw


H0 == The defective % does not varies by centre

H1 == The defective % does varies by centre

In [7]:
data = pd.read_csv("Costomer+OrderForm.csv")

In [42]:
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [45]:
stats.chi2_contingency([data["Phillippines"].value_counts(),data["Indonesia"].value_counts(),data["Malta"].value_counts(),data["India"].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

p-value is 0.28

0.2 > 0.05

Accept H0.


Defective % doesnot varies by center.